gnucash/gnome/business-urls.c

```diff
@@ -168,78 +168,77 @@ jobCB (const char *location, const char *label,
   {                                                                     \
     result->error_message =                                             \
       g_strdup_printf (_("No such owner entity: %s"), location);        \
-    return FALSE;                                                       \
+    show_report = FALSE;                                                \
   }
 
+/* parses a string "user=john&pass=smith&age=41" into a string-keyed
+   GHashTable. String must not contain non-ASCII chars. Duplicate keys
+   will be ignored. */
+static GHashTable *parse_parameters (const gchar *parms)
+{
+    GHashTable *rethash;
+    gchar *query, *p;
+
+    rethash = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, NULL);
```

Comment:
Both the key and value are `g_strdup`ed so both need to be `g_free`d: 
```C
rethash = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_free);
```

Your alternative of freeing `ownerptr` and `acctptr` assumes that those are the only values in the hashtable. Leave the ownership with the hashtable and you don't have to worry about that, nor about freeing them at the end of the function.